docker-compose/otel/docker-compose.yaml

```diff
@@ -0,0 +1,65 @@
+version: '2.4'
+services:
+  otel-collector:
```

Comment:
it looks like we have included the health check extension

we can mimic the same probe in docker-compose https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/templates/_pod.tpl#L46-L52 to perform health check

Reply:
This is kind of nontrivial actually, I think we need to build a custom collector image that can do the healthcheck without installing curl/wget (it's not included)

Reply:
> This is kind of nontrivial actually, I think we need to build a custom collector image that can do the healthcheck without installing curl/wget (it's not included)

ughh right…

now I remember docker compose health check is a joke 🫤 (we tried to do something similar for cloud sql proxy in MI but turns out it’s impossible for the same reason)

bundling wget curl doesn’t feel right to me since I heard we’re trying to remove them from our images?

let’s save health check for k8s distro

Reply:
> bundling wget curl doesn’t feel right to me since I heard we’re trying to remove them from our images?

Exactly what I was thinking - I tried to add something here but it didn't work: https://github.com/sourcegraph/sourcegraph/pull/39002/commits/e8bf8dff3186752cad502f0abc6760e02200c187 something to do with users that's beyond me, the binary is not available no matter how I spin it :lolsob:

I think I'll just leave healthchecks out of this for now